### ISSUE TITLE
Proxy auth URL encoding broken

### ISSUE THREAD

**Issue by jrjbertram** [websocket-client/websocket-client#239]
In 0.35.0, I have a very long proxy auth password and found that the base64encode function was inserting a newline into the encoded proxy string. 

I fixed the code as follows in my local copy in _http.py:tunnel... thought I would pass it on to you.

```
    #encoded_str = base64encode(auth_str.encode()).strip().decode()
    encoded_str = base64.b64encode(auth_str.encode()).strip().decode()
    print "encoded str: \"%s\"" % encoded_str
    connect_header += "Proxy-Authorization: Basic %s\r\n" % encoded_str
```

(this also required changes to import base64)

Also, I noticed in this repo there's no 0.35.0 tag/release... friendly observation in case you forgot to create it on last release, maybe?

Thank you!
- Josh.

**Comment by uriesk** [comment]
Thank you very much for this!
I had the same issue (2.5 years later now) and could fix it like that.

**Comment by engn33r** [comment]
This was fixed by PR #587
